R/jaspHistogram.R

```diff
@@ -0,0 +1,108 @@
+#' @title jaspHistogram
+#'
+#' @param x, numeric, the data to show a histogram for
+#' @param xName, string, the title on the x-axis
+#' @param rugs, logical, should rugs be shown on the x-axis?
+#' @param displayDensity, logical, should a density be superimposed on the plot?
+#' @param binWidthType, string, type of bindwidth, matches the string values that can be passed to \code{\link[graphics]{hist}}.
+#' @param numberOfBins, if binWidthType is set to "manual", this value determines the number of bins.
+#' @param densityLineWidth, the line width of the superimposed density.
+#' @example inst/examples/ex-jaspHistogram.R
+#' @export
+jaspHistogram <- function(
+  x, xName,
+  rugs = FALSE,
+  displayDensity = FALSE,
+  binWidthType = c("doane", "fd", "scott", "sturges", "manual"),
+  numberOfBins = NA,
+  densityLineWidth = 1) {
+
+  binWidthType <- match.arg(binWidthType)
+  x <- stats::na.omit(as.numeric(x))
+
+  if (binWidthType == "doane") {
+
+    # https://en.wikipedia.org/wiki/Histogram#Doane's_formula
+    sigma.g1 <- sqrt((6*(length(x) - 2)) / ((length(x) + 1)*(length(x) + 3)))
+    g1 <- mean(abs(x)^3)
+    k <- 1 + log2(length(x)) + log2(1 + (g1 / sigma.g1))
+    binWidthType <- k
+
+  } else if (binWidthType == "fd" && grDevices::nclass.FD(x) > 10000) { # FD-method will produce extreme number of bins and crash ggplot, mention this in footnote
+
+    binWidthType <- 10000
+
+  } else if (binWidthType == "manual") {
```

Comment:
```suggestion
  } else if (binWidthType == "manual") {
  if(is.na(numberOfBins)) stop("numberOfBins argument must be specified when a binWidthType == 'manual'.")
```
Just fail early with an informative message